custom/icds/translations/integrations/transifex.py

```diff
@@ -0,0 +1,200 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+import polib
+import datetime
+import os
+
+from django.utils.translation import ugettext as _
+from django.conf import settings
+
+from collections import namedtuple, OrderedDict
+from corehq.apps.app_manager import id_strings
+from custom.icds.translations.integrations.client import TransifexApiClient
+
+Translation = namedtuple('Translation', 'key translation occurrences')
+Unique_ID = namedtuple('UniqueID', 'type id')
+
+
+class Transifex():
+    def __init__(self, domain, app_id, source_lang, target_lang=None, version=None, lang_prefix='default_'):
+        """
+        :param domain: domain name
+        :param app_id: id of the app to be used
+        :param source_lang: source lang code like en
+        :param target_lang: target lang code like hin
+        :param version: version of the app like 10475
+        :param lang_prefix: prefix if other than "default_"
+        """
+        self.app_id = app_id
+        self.domain = domain
+        self.version = int(version)
+        self.source_lang = source_lang
+        self.target_lang = target_lang
+        self.lang_prefix = lang_prefix
+        if not self.target_lang:
+            self.target_lang = self.source_lang
+        self.app_id_to_build = None
+        self.translations = OrderedDict()
+        self.headers = dict()  # headers for each sheet name
+        self.sheet_name_to_module_or_form_type_and_id = dict()
+        self.generated_files = list()
+
+    def find_build_id(self):
+        # find build id if version specified
+        if self.version:
+            from corehq.apps.app_manager.dbaccessors import get_all_built_app_ids_and_versions
+            built_app_ids = get_all_built_app_ids_and_versions(self.domain, self.app_id)
+            for app_built_version in built_app_ids:
+                if app_built_version.version == self.version:
+                    self.app_id_to_build = app_built_version.build_id
+                    break
+            if not self.app_id_to_build:
+                raise Exception("Build for version requested not found")
+        else:
+            self.app_id_to_build = self.app_id
+
+    def translation_data(self, app):
+        # get the translations data
+        from corehq.apps.app_manager.app_translations.app_translations import expected_bulk_app_sheet_rows
+        # simply the rows of data per sheet name
+        rows = expected_bulk_app_sheet_rows(app)
+
+        # get the translation data headers
+        from corehq.apps.app_manager.app_translations.app_translations import expected_bulk_app_sheet_headers
+        for header_row in expected_bulk_app_sheet_headers(app):
+            self.headers[header_row[0]] = header_row[1]
+        self.set_sheet_name_to_module_or_form_mapping(rows[u'Modules_and_forms'])
+        return rows
```

Comment:
this might be tricky. Setting headers and that mapping relies to have the `rows` data populated. so its like a series of steps that happen. Let me try to improve this while i move this out to a `PoFilesUtility`.